### PR TITLE
Shader naming convention

### PR DESCRIPTION
**Pull request by jamesmporter**
Here's an idea for a naming convention for the shader attributes. I have opened this as a draft PR so you can give your feedback on the convention so far.

Addresses #1396

### CODE REVIEW

File: packages/d3fc-webgl/src/series/glBar.js [hunk 1]
```diff
@@ -66,9 +66,9 @@ export default () => {
         program
             .buffers()
             .attribute('aXValue', xValueAttribute)
-            .attribute('aYValue', yValueAttribute)
-            .attribute('aY0Value', y0ValueAttribute)
-            .attribute('aWidthValue', widthValueAttribute)
+            .attribute('aXWidth', widthValueAttribute)
```
Comment (chrisprice) [inline packages/d3fc-webgl/src/series/glBar.js]: Does series call this bandwidth?

File: packages/d3fc-webgl/src/series/glArea.js [hunk 2]
```diff
@@ -52,8 +52,8 @@ export default () => {
         .attribute('aXPrevValue', xPreviousValueAttribute)
         .attribute('aYValue', yValueAttribute)
         .attribute('aYPrevValue', yPreviousValueAttribute)
-        .attribute('aY0Value', y0ValueAttribute)
-        .attribute('aY0PrevValue', y0PreviousValueAttribute)
+        .attribute('aYBase', y0ValueAttribute)
+        .attribute('aYPrevBase', y0PreviousValueAttribute)
```
Comment (chrisprice) [inline packages/d3fc-webgl/src/series/glArea.js]: Maybe it's time to go the whole hog and use main/cross/base/orient?

File: packages/d3fc-webgl/src/series/glBar.js (hunk 1 above)
Comment (chrisprice) [reply packages/d3fc-webgl/src/series/glBar.js]: I'm not keen on the x/y prefixing because we'll ultimately need to support orientation

**Comment by jamesmporter** [comment]
The cross has stroke-width instead of line-width.
@chrisprice Do you know why this is?

File: packages/d3fc-webgl/src/series/glBar.js [hunk 3]
```diff
@@ -66,9 +66,9 @@ export default () => {
         program
             .buffers()
             .attribute('aXValue', xValueAttribute)
```
Comment (chrisprice) [inline packages/d3fc-webgl/src/series/glBar.js]: How do you feel about the (in)consistency of when we use the value suffix?

**Comment by chrisprice** [comment]
> The cross has stroke-width instead of line-width.
> @chrisprice Do you know why this is?

I think it's a by-product of svg using `stroke` and canvas using `line`. We should pick one and stick with it. I'd recommend `stroke`

File: packages/d3fc-webgl/src/series/glBar.js (hunk 3 above)
Comment (chrisprice) [reply packages/d3fc-webgl/src/series/glBar.js]: https://d3fc.io/api/series-api.html#ohlc

**Comment by jamesmporter** [comment]
I'd like to differ from the [D3FC docs](https://d3fc.io/api/series-api.html#seriesPoint_size) on the point of `aSize`. The shaders will be clearer if we use `aArea`.

File: packages/d3fc-webgl/src/series/glBar.js (hunk 1 above)
Comment (jamesmporter) [reply packages/d3fc-webgl/src/series/glBar.js]: I've changed this to cross and main

File: packages/d3fc-webgl/src/series/glArea.js (hunk 2 above)
Comment (jamesmporter) [reply packages/d3fc-webgl/src/series/glArea.js]: I've implemented this.

File: packages/d3fc-webgl/src/series/glBar.js (hunk 3 above)
Comment (jamesmporter) [reply packages/d3fc-webgl/src/series/glBar.js]: Implemented.

**Comment by jamesmporter** [comment]
I've renamed all instances of `uLineWidth` to `uStrokeWidth`.

**Comment by chrisprice** [comment]
> I'd like to differ from the [D3FC docs](https://d3fc.io/api/series-api.html#seriesPoint_size) on the point of `aSize`. The shaders will be clearer if we use `aArea`.

Yea I agree it's weird but we didn't come up with that one - https://github.com/d3/d3-shape#symbol_size

File: packages/d3fc-webgl/src/series/glPoint.js [hunk 4]
```diff
@@ -19,9 +19,9 @@ export default () => {
 
     program
         .buffers()
-        .attribute('aXValue', xValueAttribute)
-        .attribute('aYValue', yValueAttribute)
-        .attribute('aSize', sizeAttribute);
+        .attribute('aCrossValue', xValueAttribute)
+        .attribute('aMainValue', yValueAttribute)
+        .attribute('aArea', sizeAttribute);
```
Comment (chrisprice) [inline packages/d3fc-webgl/src/series/glPoint.js]: Please can you make this `aSize` and then I think we're good

**Review by chrisprice — CHANGES_REQUESTED** [review]

File: packages/d3fc-webgl/src/series/glPoint.js (hunk 4 above)
Comment (jamesmporter) [reply packages/d3fc-webgl/src/series/glPoint.js]: Done.

**Review by chrisprice — APPROVED** [review]

**Pull request merged**